File: 01/main.py

file = open('input.txt', 'r')
lines = file.readlines()

elves = {
    0: 0,
}

elf_index = 0

for line in lines:
    line = line.strip()
    if line == '':
        elf_index += 1
        elves[elf_index] = 0
    else:
        elves[elf_index] += int(line)

sorted_values = sorted(elves.values())

print(f'Highest: {sorted_values[len(sorted_values)-1]}')

top_3_total = sorted_values[len(sorted_values)-1] + \
    sorted_values[len(sorted_values)-2] + \
    sorted_values[len(sorted_values)-3]

print(f'Top 3 total: {top_3_total}')

File: 01/main.php
<?php

$content = file_get_contents('input.txt');
$content = explode("\n", $content);

$elves = [
	0 => 0,
];
$elf_no = 0;

foreach ($content as $line)
{
	if (trim($line) === '') {
		$elf_no++;
		$elves[$elf_no] = 0;
		continue;
	}

	$elves[$elf_no] += $line;
}

sort($elves);

echo 'Highest single calorie value: ' . max($elves) . "\n";

$top_3_total = 0;

for ($i = 0; $i < 3; $i++)
{
	$top_3_total += array_pop($elves);
}

echo 'Top 3 total: ' . $top_3_total . "\n";
